hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/RecordFileServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package com.hedera.mirror.web3.service;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import com.hedera.mirror.rest.model.Block;
+import com.hedera.mirror.web3.Web3IntegrationTest;
+import com.hedera.mirror.web3.exception.BlockNumberOutOfRangeException;
+import com.hedera.mirror.web3.viewmodel.BlockType;
+import lombok.RequiredArgsConstructor;
+import org.junit.jupiter.api.Test;
+
+@RequiredArgsConstructor
+public class RecordFileServiceTest extends Web3IntegrationTest {
+    private final RecordFileService recordFileService;
+
+    @Test
+    void testFindByTimestamp() {
+        var timestamp = domainBuilder.timestamp();
+        var recordFile = domainBuilder.recordFile()
+                .customize(e -> e.consensusEnd(timestamp))
+                .persist();
+        assertThat(recordFileService.findByTimestamp(timestamp)).get().isEqualTo(recordFile);
```

Comment:
```suggestion
        assertThat(recordFileService.findByTimestamp(timestamp)).contains(recordFile);
```

this applies all other similar places